app/controllers/registration/registration.controller.test.js

```diff
@@ -69,6 +70,114 @@ describe('Registration', () => {
       sinon.assert.notCalled(res.redirect)
     })
   })
+
+  describe('submit the password page', () => {
+    it('should call next with an error when the registration cookie is not set', async () => {
+      delete req.register_invite
+      await registrationController.submitPasswordPage(req, res, next)
+
+      sinon.assert.calledWith(next, sinon.match.instanceOf(RegistrationSessionMissingError))
+      sinon.assert.notCalled(res.render)
+      sinon.assert.notCalled(res.redirect)
+    })
+
+    it('should render the password page with an error when both password fields are empty', async () => {
+      req.body = {
+        password: '',
+        'repeat-password': ''
+      }
+      await registrationController.submitPasswordPage(req, res, next)
+
+      sinon.assert.calledWith(res.render, 'registration/password', {
```

Comment:
Test coverage looks great but as the comment above I wonder if these should be covering a simple service and then the controller just checks once that the appropriate error object is passed through

Reply:
https://github.com/alphagov/pay-selfservice/pull/3622#discussion_r1025298260